src/sentry.ts

```diff
@@ -178,7 +185,45 @@ export default class SentryService {
   async getFilenamesFromDirectory(): Promise<any[]> {
     const path = this.app.config?.getConfigDir();
     const addr = fs.readdirSync(path, { withFileTypes: true });
-    return addr.filter(f => f.isFile()).map(f => ({ name: f.name }));
+    return addr.filter(f => f.isFile()).map(f => ({ name: f.name })) || [];
+  }
+
+  async readCommandHistoryFromFileSystem(): Promise<any[]> {
+    let command_history = [];
+    if (fs.existsSync(this.sentry_history_file_path)) {
+      try {
+        command_history = await JSON.parse(fs.readFileSync(this.sentry_history_file_path).toString()) || [];
+      } catch {
+        // If json.parse fails, don't error.
+      }
+    }
+    return command_history;
+  }
+
+  async writeCommandHistoryToFileSystem(): Promise<void> {
+    const scope = Sentry.getCurrentHub()?.getScope();
+    const remove_keys = new Set(['plugins', 'pjson', 'oauth_client_id', 'credentials', '_auth_result', 'Authorization']);
+
+    // remove empty objects, empty strings, or key/values from the remove_keys list.
+    let current_output = JSON.parse(JSON.stringify(scope as any, (key, value) => {
```

Comment:
Rather than stringifying then parsing `scope`, why not just iterate over key/value pairs and set values to remove equal to undefined?